### ISSUE TITLE
Refresh rate

### ISSUE THREAD

**Issue by mjjzf** [h3poteto/whalebird-desktop#242]
I am running Whalebird on my work Windows 10 machine.
The time line does not seem to update - if I change from Home to Notification and back to Home, Whalebird loads the updated feed, otherwise not. I can not seem to find a setting for it.
Am I missing something?

**Comment by h3poteto** [comment]
Whalebird is running user stream in the background. Normally, Home timeline is updated with streaming; therefore, refresh rate is not exist and can not be controlled.

Where instance do you create an account? I will try on the same instance.

**Comment by mjjzf** [comment]
I am on mstdn.io.

**Comment by mjjzf** [comment]
Sorry, closed the bug on accident.

**Comment by h3poteto** [comment]
@mjjzf That! I have an account in mstdn.io too, and I can not receive statuses in stream.
I was already aware, and I tried following code:

```javascript
let client = new Mastodon(
      {
        access_token: accessToken,
        api_url: 'https://mstdn.io/api/v1'
      }
)

let listener = client.stream('/streaming/user')
listener.on('message', (msg) => {
  console.log(msg)
})

this.listener.on('error', (err) => {
  console.log(err)
})
```

But it does not receive anything, and it does not output any error.

I will ask the administrator of the mstdn.io about this.

**Comment by h3poteto** [comment]
https://mstdn.io/@angristan/99886319166339772

Sorry, I couldn't know new informations.

**Comment by gfrenoy** [comment]
Got the same issue with my instance ; no new toot is popping in.  Still, when using the default browser client, I'm receiving the new toots automatically ; I'll need to dig into the mastodon instance configuration to understand why.

Maybe an idea : have a "manual" refresh menu entry ?  Or when you scroll up (like on a smartphone app) ?

**Comment by gfrenoy** [comment]
Ok so let's first try to understand why that streaming update does not work.

**Comment by gfrenoy** [comment]
Got the streaming API working for me !

The root cause is that the default apache configuration proposed in the [documentation](https://github.com/tootsuite/documentation/blob/master/Running-Mastodon/Alternatives.md) does not manage the streaming on HTTP request.  See here for details : https://github.com/tootsuite/documentation/pull/439

I updated my configuration as suggested [here](https://github.com/tootsuite/documentation/pull/439/commits/f250b9d990e9bb801bb512b70cf142ac11983771) and I now receive the notification in Whalebird :)

**Comment by h3poteto** [comment]
Thank you, I see. 
But I am not an administrator of mstdn.io, so I have to ask the administrator to change apache configuration.

**Comment by SinaCutie** [comment]
I can go to `Preferences` page and then close out of it to refresh my timeline manually. Perhaps a manual refresh would just refresh the main view the way opening and closing Preferences does?

It is the only way I can make Whalebird usable for reading.

**Comment by gfrenoy** [comment]
Actually, the issue is on the server side.  To check that you can try with the code in comment https://github.com/h3poteto/whalebird-desktop/issues/242#issuecomment-382710618.

After I fixed the server, I did not need to refresh the timeline "manually", Whalebird is doing it automagically... 

Now, in my case, I'm managing the server side as well so I could fix the issue but what about users who cannot control the server configuration ?

At least a warning should be displayed if the server does not support the streaming API correctly.

**Comment by h3poteto** [comment]
I added reload function, and reload button.
https://github.com/h3poteto/whalebird-desktop/pull/381

**Comment by h3poteto** [comment]
I fixed it in #786 .
If the timeline does not auto refresh, please try websocket streaming.
I can stream with websocket in mstdn.io.
